### PR TITLE
Performance: LogCommits to fetch ranges only when needed

### LINKED ISSUE #5224: Bug lakectl blame hangs without output for long time

### PR DESCRIPTION
**Pull request by itaiad200**
closes #5224

### CODE REVIEW

File: pkg/graveler/graveler.go [hunk 1]
```diff
@@ -1482,6 +1488,62 @@ func (g *Graveler) GetByCommitID(ctx context.Context, repository *RepositoryReco
...
+
+	if !lFound && !rFound {
+		// no range matching the key exist in both commits
+		return true, nil
+	}
+	if lRangeID == rRangeID {
+		// It's the same range - the value of the key is identical in both
+		return true, nil
+	}
+
```
Comment (N-o-Z) [inline pkg/graveler/graveler.go:1521]:
```suggestion
if lFound != rFound {
  return false, nil
}

```

File: pkg/graveler/graveler.go [hunk 2]
```diff
@@ -1482,6 +1488,62 @@ func (g *Graveler) GetByCommitID(ctx context.Context, repository *RepositoryReco
...
+	switch {
+	case !lFound && !rFound:
+		// not found in both commits
+		return true, nil
+	case !lFound || !rFound:
+		// found in one commit but not in the other
+		return false, nil
+	}
```
Comment (N-o-Z) [inline pkg/graveler/graveler.go:1541]: I think you don't need this switch case once you add the 'if' suggested above

**Review by N-o-Z — CHANGES_REQUESTED** [review]
Thanks @itaiad200 !
General note: Didn't quite understand why we removed the caching mechanism
I have some suggestion - overall looks good!

**Review by itaiad200 — COMMENTED** [review]
Bringing back `storeGetCache` due to popular demand

File: pkg/graveler/committed/meta_range_manager.go [hunk 3]
```diff
@@ -50,21 +50,9 @@ func (m *metaRangeManager) Exists(ctx context.Context, ns graveler.StorageNamesp
 // GetValue finds the matching graveler.ValueRecord in the MetaRange with the rangeID
 func (m *metaRangeManager) GetValue(ctx context.Context, ns graveler.StorageNamespace, id graveler.MetaRangeID, key graveler.Key) (*graveler.ValueRecord, error) {
```
Comment (N-o-Z) [inline pkg/graveler/committed/meta_range_manager.go:51]: I don't think this is being used anywhere

**Review by N-o-Z — CHANGES_REQUESTED** [review]
Some added comments

File: .github/workflows/docker-publish-exp-image.yaml [hunk 4]
```diff
@@ -64,9 +64,9 @@ jobs:
         uses: docker/build-push-action@v3
         with:
           context: .
-          target: lakefs-plugins
+          target: lakefs-lakectl
           push: true
-          platforms: linux/amd64,linux/arm64,darwin/amd64,darwin/arm64
```
Comment (N-o-Z) [inline .github/workflows/docker-publish-exp-image.yaml:69]: I think you want to revert this

File: pkg/catalog/catalog.go [hunk 5]
```diff
@@ -1106,7 +1106,7 @@ func (c *Catalog) listCommitsWithPaths(ctx context.Context, repository *graveler
 	if len(params.PathList) == 0 {
 		return nil, false, fmt.Errorf("%w: list commits without paths", graveler.ErrInvalid)
 	}
-	// commit/key to value cache - helps when fetching the same commit/key while processing parent commits
+
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply pkg/catalog/catalog.go:1109]: It is. Done.

File: pkg/catalog/catalog.go [hunk 6]
```diff
@@ -1338,6 +1338,31 @@ func (c *Catalog) checkPathListInCommit(ctx context.Context, repository *gravele
 				return false, err
 			}
 		} else {
+			// check if the key exists in both commits
+
```
[the remark this answers is not in the document]
Comment (itaiad200) [reply pkg/catalog/catalog.go]: Done.

File: pkg/graveler/committed/meta_range_manager.go (hunk 3 above)
Comment (itaiad200) [reply pkg/graveler/committed/meta_range_manager.go:51]: Let's remove in a separate PR?

**Review by N-o-Z — APPROVED** [review]
Thanks - let's just open the issue to review the metarange manager and remove unused code there

**Pull request merged**